airflow/sensors/filesystem.py

```diff
@@ -75,3 +92,20 @@ def poke(self, context: Context):
                 if files:
                     return True
         return False
+
+    def execute(self, context: Context) -> None:
+        if self.deferrable and not self.poke(context=context):
+            self.defer(
+                timeout=datetime.timedelta(seconds=self.timeout),
+                trigger=FileTrigger(
+                    filepath=self.path,
+                    recursive=self.recursive,
+                    poke_interval=self.poke_interval,
+                ),
+                method_name="execute_complete",
+            )
```

Comment:
```suggestion
        if not self.deferrable:
            super().execute(context=context)
        elif not self.poke(context=context):
            self.defer(
                timeout=datetime.timedelta(seconds=self.timeout),
                trigger=FileTrigger(
                    filepath=self.path,
                    recursive=self.recursive,
                    poke_interval=self.poke_interval,
                ),
                method_name="execute_complete",
            )
```
```suggestion
        if self.deferrable and not self.poke(context=context):
            self.defer(
                timeout=datetime.timedelta(seconds=self.timeout),
                trigger=FileTrigger(
                    filepath=self.path,
                    recursive=self.recursive,
                    poke_interval=self.poke_interval,
                ),
                method_name="execute_complete",
            )
```

Should we try something like this instead? Otherwise, non-deferred mode might only poke once

Reply:
makes sense 👍